core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala

```diff
@@ -17,49 +17,100 @@
+        H2O.shutdown(-1)
       }
+      throw new RuntimeException("Cloud size " + discoveredSize + " under " + expectedSize);
     }
-    H2O.CLOUD.size()
-  }
 
-  /**
-    * Get arguments for H2O client
-    *
-    * @return array of H2O client arguments.
-    */
-  override def getH2OClientArgs(conf: H2OConf): Seq[String] = {
-    val builder = new ArgumentBuilder()
-      .add(super.getH2OClientArgs(conf))
-      .addIf("-watchdog_client", conf.isAutoClusterStartUsed)
+    // Register web API for client
+    RestAPIManager(hc).registerAll()
+    H2O.startServingRestApi()
+  }
 
-    if (conf.isAutoClusterStartUsed) {
-      builder.add("-flatfile", new File(conf.externalBackendFlatFileName().get).getAbsolutePath)
+  protected def startUnhealthyStateKillThread(conf: H2OConf): Option[Thread] = {
+    if (conf.isKillOnUnhealthyClusterEnabled) {
+      val thread = new Thread {
+        override def run(): Unit = {
+          while (true) {
+            Thread.sleep(conf.killOnUnhealthyClusterInterval)
+            if (!H2O.CLOUD.healthy() && conf.isKillOnUnhealthyClusterEnabled) {
+              Log.err("Exiting! External H2O cluster not healthy!!")
+              H2O.shutdown(-1)
+            }
+          }
+        }
+      }
+      thread.setDaemon(true)
+      thread.start()
+      Some(thread)
     } else {
-      builder.add("-flatfile", conf.h2oCluster.map(clusterStr => saveFlatFileAsFile(clusterStr).getAbsolutePath))
+      None
     }
-    builder.buildArgs()
+  }
 
+  protected def startUnhealthyStateCheckThread(conf: H2OConf): Option[Thread] = {
+    val thread = new Thread {
+      override def run(): Unit = {
+        while (true) {
+          Thread.sleep(conf.healthCheckInterval)
+          if (!H2O.CLOUD.healthy()) {
+            Log.err("External H2O cluster not healthy!!")
+          }
+        }
+      }
+    }
+    thread.setDaemon(true)
+    thread.start()
+    Some(Thread)
   }
 
+  protected def verifyH2OClientCloudUp(conf: H2OConf): Unit = {
+    val cloudMembers = H2O.CLOUD.members()
+    if (cloudMembers.isEmpty) {
+      if (conf.isManualClusterStartUsed) {
+        throw new H2OClusterNotRunning(
+          s"""
+             |External H2O cluster is not running or could not be connected to. Provided configuration:
+             |  cluster name            : ${conf.cloudName.get}
+             |  cluster representative  : ${conf.h2oCluster.getOrElse("Using multi-cast discovery!")}
+             |  cluster start timeout   : ${conf.clusterStartTimeout} sec
+             |
+             |It is possible that in case you provided only the cluster name, h2o is not able to cloud up
+             |because multi-cast communication is limited in your network. In that case, please consider starting the
+             |external H2O cluster with flatfile and set the following configuration '${
+            ExternalBackendConf.
+              PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1
```

Comment:
nit: `ExternalBackendConf.PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1` could be on one line.

Reply:
👍